### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by kaustubh-nair**
Part of #304 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

[//]: # (To mark checkboxe write 'x' within the square brackets)

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -33,12 +33,11 @@ def create
 
   def update
     @comment = Comment.find params[:id]
-
     if logged_in? && current_user.can_edit?(@comment)
-      Comment.update(@comment.id, :body => params[:comment][:body])
```
Comment (kaunair) [inline app/controllers/comments_controller.rb]: This line was not updating the comment and the tests were failing. That's why I had to change it

**Comment by kaunair** [comment]
@jywarren I just noticed the repo_token for coveralls is not loading. I wonder why that's happening?
It is working for some prs like this https://github.com/publiclab/mapknitter/pull/462
but not for some :thinking:

**Comment by kaunair** [comment]
Looks like the token is loading for push builds but not for pr builds

**Comment by kaunair** [comment]
Added some more tests..

**Comment by kaunair** [comment]
Hey @jywarren , please review this!
Thanks!

**Comment by jywarren** [comment]
Hmm, i don't know -- and what's going on with Travis tests here? Are they stuck? They had some issues at Travis last week i think... could it be related?

**Comment by jywarren** [comment]
And... thank you!

**Comment by kaunair** [comment]
The problem is that travis does not expose environment variables for pull requests from external repositories. You can read about it here https://docs.travis-ci.com/user/pull-requests#pull-requests-and-security-restrictions
This is a really restricting constraint. I'm trying to find a good workaround for this.

**Comment by kaunair** [comment]
The simplest way would be the make the REPO_TOKEN env variable public. But someone else can use it to send test coverage reports to coveralls. So I guess we can eliminate this approach?

**Comment by kaunair** [comment]
But their security concern also makes sense.
If an external pr has access to the environment variable, they could just send some malicious code that could print the variable out. So giving external prs access to environment variables is essentially the same as making it public. We might as well put the environment variable in our repository then?
Thoughts? @jywarren

**Comment by jywarren** [comment]
Hmm, so how do we do this in plots2, where it is hooked up?
https://github.com/publiclab/plots2/tree/master/.travis.yml

Thank you for thinking this through!

**Comment by kaunair** [comment]
Is it working for plots2?
I just checked some recent pr builds and it looks like travis isn't sending coverage report for those prs
Like for example
https://travis-ci.org/publiclab/plots2/builds/512970864#L3498
and https://travis-ci.org/publiclab/plots2/builds/514791356#L4206
It works for internal prs like this one though
https://travis-ci.org/publiclab/plots2/builds/511213184#L3486

**Comment by kaunair** [comment]
Coverage reports on coveralls are mostly for push builds.
Not for pull requests I think

**Comment by kaustubh-nair** [comment]
@jywarren If you think the tests are fine then should we merge this? I'll send more tests in subsequent prs.

**Comment by kaustubh-nair** [comment]
And what should be done about the travis issue?
Should I open a new issue to discuss it there?

**Comment by jywarren** [comment]
Yes, let's do! Thank you. And yes i'll merge this. Thanks so much! Heads up @cesswairimu who is also looking at improving testing (to coordinate!)

**Comment by kaustubh-nair** [comment]
Awesome, thanks 
@cesswairimu Please mention which sections you are improving on(whenever you do) so that we don't end up working on the same thing. :smiley:

**Comment by cesswairimu** [comment]
Yeah great idea, I had implemented comments_controller tests before I saw your pr :woman_facepalming:, I am currently implementing tag, images controller tests and figuring out the sessions_controller tests will open a `wip` too. Thanks @kaustubh-nair

**Pull request merged**